src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js

```diff
@@ -69,18 +70,78 @@ class ActivityFeed extends React.Component<Props, State> {
     approvalCommentFormSubmitHandler = (): void => this.setState({ isInputOpen: false });
 
     /**
-     * Creates a comment
+     * Add a placeholder pending feed item.
      *
-     * @param {string} text - Comment text
-     * @param {boolean} hasMention - If this comment contains an at mention
+     * @param {Object} itemBase - Base properties for item to be added to the feed as pending.
+     * @return {void}
+     */
+    addPendingItem = (itemBase: Object): void => {
+        const { currentUser } = this.props;
+        // create a placeholder pending feed item
+        const date = new Date().toISOString();
+        const feedItem = {
+            created_at: date,
+            created_by: currentUser,
+            modified_at: date,
+            isPending: true,
+            ...itemBase
+        };
+
+        const { feedItems } = this.state;
+        this.setState({ feedItems: [feedItem, ...feedItems] });
+    };
+
+    /**
+     * Replace a feed item with new feed item data.
+     *
+     * @param {Comment | Task} feedItem - API returned feed item data.
+     * @param {string} uuid - Unique ID of the feed item to replace.
+     * @return {void}
+     */
+    updateFeedItem = (feedItem: Comment | Task, uuid: string): void => {
+        let itemIndex = null;
+        const { feedItems } = this.state;
+        const newFeedItems = feedItems.slice();
+
+        newFeedItems.find((item, index) => {
+            if (item.id === uuid) {
+                itemIndex = index;
+                return true;
+            }
+            return false;
+        });
+
+        // Replace item in the feed items or set as most recent item.
+        if (itemIndex !== null) {
+            newFeedItems[itemIndex] = feedItem;
+            this.setState({ feedItems: newFeedItems });
+        }
+    };
+
+    /**
+     * Create a comment, and make a pending item to be replaced once the API is successful.
+     *
+     * @param {any} args - Data returned by the Comment component on comment creation.
      * @return {void}
      */
     createComment = ({ text, hasMention }: { text: string, hasMention: boolean }): void => {
-        // create a placeholder pending comment
-        // create actual comment and send to Box V2 api
-        // call user passed in handlers.comments.create, if it exists
-        const createComment = getProp(this.props, 'handlers.comments.create', noop);
-        createComment(text, hasMention);
+        const uuid = uniqueId('comment_');
+        const comment = {
+            id: uuid,
+            tagged_message: text,
+            type: 'comment'
+        };
+
+        this.addPendingItem(comment);
+
+        const createComment = getProp(this.props, 'handlers.comments.create', Promise.resolve({}));
+        createComment(text, hasMention).then((commentData) => {
```

Comment:
Am doing something similar right now with updating tasks. What do you think of this pattern instead, since we are already using callbacks?

```
    onTaskUpdate = (
        taskId: string,
        message: string,
        successCallback: (task: Task) => void = noop,
        errorCallback: (e: Error) => void = noop,
        dueAt?: string
    ) => {
        const { file } = this.state;

        if (!file) {
            throw getBadItemError();
        }

        this.api.getTasksAPI(false).updateTask({
            file,
            taskId,
            message,
            dueAt,
            successCallback: (task: Task) => {
                successCallback(task);
                this.onTaskUpdateSuccessCallback(task);
            },
            errorCallback: (e: Error) => {
                errorCallback(e);
                this.errorCallback(e);
            }
        });
    };
```